java/dataset/src/test/java/org/apache/arrow/dataset/file/TestSingleFileDataset.java

```diff
@@ -0,0 +1,237 @@
+
+  public static final String AVRO_SCHEMA_USER = "user.avsc";
+
+  @Test
+  public void testParquetRead() throws Exception {
+    ParquetWriteSupport writeSupport = new ParquetWriteSupport(AVRO_SCHEMA_USER, TMP.newFolder());
+    GenericRecord record = new GenericData.Record(writeSupport.getAvroSchema());
+    record.put("id", 1);
+    record.put("name", "a");
+    writeSupport.writeRecord(record);
+    writeSupport.close();
+
+    SingleFileDatasetFactory factory = new SingleFileDatasetFactory(rootAllocator(),
+        FileFormat.PARQUET, FileSystem.LOCAL, writeSupport.getOutputFile());
+    ScanOptions options = new ScanOptions(new String[0], Filter.EMPTY, 100);
+
+    Schema schema = inferResultSchemaFromFactory(factory, options);
+    assertEquals(2, schema.getFields().size());
+    assertEquals("id", schema.getFields().get(0).getName());
+    assertEquals("name", schema.getFields().get(1).getName());
+    assertEquals(Types.MinorType.INT.getType(), schema.getFields().get(0).getType());
+    assertEquals(Types.MinorType.VARCHAR.getType(), schema.getFields().get(1).getType());
+
+    assertSingleTaskProduced(factory, options);
+
+    List<ArrowRecordBatch> datum = collectResultFromFactory(factory, options);
+    assertEquals(1, datum.size());
+    checkParquetReadResult(schema, Collections.singletonList(record), datum);
+    AutoCloseables.close(datum);
+  }
+
+  @Test
+  public void testParquetProjector() throws Exception {
+    ParquetWriteSupport writeSupport = new ParquetWriteSupport(AVRO_SCHEMA_USER, TMP.newFolder());
+    GenericRecord record = new GenericData.Record(writeSupport.getAvroSchema());
+    record.put("id", 1);
+    record.put("name", "a");
+    writeSupport.writeRecord(record);
+    writeSupport.close();
+
+    SingleFileDatasetFactory factory = new SingleFileDatasetFactory(rootAllocator(),
+        FileFormat.PARQUET, FileSystem.LOCAL, writeSupport.getOutputFile());
+    ScanOptions options = new ScanOptions(new String[]{"id"}, Filter.EMPTY, 100);
+
+    Schema schema = inferResultSchemaFromFactory(factory, options);
+    assertEquals(1, schema.getFields().size());
+    assertEquals("id", schema.getFields().get(0).getName());
+    assertEquals(Types.MinorType.INT.getType(), schema.getFields().get(0).getType());
+
+    assertSingleTaskProduced(factory, options);
+
+    List<ArrowRecordBatch> datum = collectResultFromFactory(factory, options);
+    assertEquals(1, datum.size());
+    org.apache.avro.Schema expectedSchema = truncateAvroSchema(writeSupport.getAvroSchema(), 0, 1);
+    checkParquetReadResult(schema,
+        Collections.singletonList(
+            new GenericRecordBuilder(
+                expectedSchema)
+                .set("id", 1)
+                .build()), datum);
+    AutoCloseables.close(datum);
+  }
+
+  @Test
+  public void testParquetFilter() throws Exception {
+    ParquetWriteSupport writeSupport = new ParquetWriteSupport(AVRO_SCHEMA_USER, TMP.newFolder());
```

Comment:
please try to break tests into: setup, execute and verify sections.  lack of spacing/comments make these hard to read.

Reply:
done. optimized with spacings